superset/views/core.py

```diff
@@ -741,6 +741,15 @@ def explore(
             if value:
                 initial_form_data = json.loads(value)
 
+        if not initial_form_data:
+            slice_id = request.args.get("slice_id")
+            if slice_id:
+                initial_form_data["slice_id"] = slice_id
+
+            dataset_id = request.args.get("dataset_id")
+            if dataset_id:
+                initial_form_data["datasource"] = f"{dataset_id}__table"
```

Comment:
simplify codes

```
        slice_id = request.args.get("slice_id")
        dataset_id = request.args.get("dataset_id")
        if not initial_form_data and not slice_id and not dataset_id:
            flash('some error message if can't get slice_id and dataset_id')

        if not initial_form_data:
            initial_form_data["slice_id"] = slice_id
            initial_form_data["datasource"] = f"{dataset_id}__table"
```

Reply:
I changed the logic a little bit to only use one or the other and present a specific message for each type 😉